test/e2e/apps/deployment.go

```diff
@@ -134,6 +141,238 @@ var _ = SIGDescribe("Deployment", func() {
+							"command": []string{"/bin/sleep", "100000"},
+						}},
+					},
+				},
+			},
+		})
+		framework.ExpectNoError(err, "failed to Marshal Deployment JSON patch")
+		_, err = f.ClientSet.AppsV1().Deployments(testNamespaceName).Patch(context.TODO(), testDeploymentName, types.StrategicMergePatchType, []byte(deploymentPatch), metav1.PatchOptions{})
+		framework.ExpectNoError(err, "failed to patch Deployment")
+
+		foundEvent = false
+		for event := range dplmtWatchChan {
+			if event.Type == watch.Modified {
+				foundEvent = true
+				break
+			}
+		}
+		framework.ExpectEqual(foundEvent, true, "failed to see scale of replicas")
+
+		ginkgo.By("waiting for Replicas to scale")
+		foundEvent = false
+		for event := range dplmtWatchChan {
+			deployment, ok := event.Object.(*appsv1.Deployment)
+			framework.ExpectEqual(ok, true, "unable to convert event.Object type")
+			if deployment.Status.AvailableReplicas == testDeploymentMinimumReplicas &&
+				deployment.Status.ReadyReplicas == testDeploymentMinimumReplicas {
+				foundEvent = true
+				break
+			}
+		}
+		framework.ExpectEqual(foundEvent, true, "failed to see scale of replicas")
+
+		ginkgo.By("listing Deployments")
+		deploymentsList, err := f.ClientSet.AppsV1().Deployments("").List(context.TODO(), metav1.ListOptions{LabelSelector: testDeploymentLabelsFlat})
+		framework.ExpectNoError(err, "failed to list Deployments")
+		foundDeployment := false
+		for _, deploymentItem := range deploymentsList.Items {
+			if deploymentItem.ObjectMeta.Name == testDeploymentName &&
+				deploymentItem.ObjectMeta.Namespace == testNamespaceName &&
+				deploymentItem.ObjectMeta.Labels["test-deployment-static"] == "true" &&
+				*deploymentItem.Spec.Replicas == testDeploymentMinimumReplicas &&
+				deploymentItem.Spec.Template.Spec.Containers[0].Image == testDeploymentPatchImage {
+				foundDeployment = true
+				break
+			}
+		}
+		framework.ExpectEqual(foundDeployment, true, "unable to find the Deployment in list", deploymentsList)
+
+		ginkgo.By("updating the DeploymentStatus")
+		testDeploymentUpdate := testDeployment
+		testDeploymentUpdate.ObjectMeta.Labels["test-deployment"] = "updated"
+		testDeploymentUpdate.Spec.Template.Spec.Containers[0].Image = testDeploymentUpdateImage
+		testDeploymentDefaultReplicasPointer := &testDeploymentDefaultReplicas
+		testDeploymentUpdate.Spec.Replicas = testDeploymentDefaultReplicasPointer
+		testDeploymentUpdateUnstructuredMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&testDeploymentUpdate)
+		framework.ExpectNoError(err, "failed to convert to unstructured")
+		testDeploymentUpdateUnstructured := unstructuredv1.Unstructured{
+			Object: testDeploymentUpdateUnstructuredMap,
+		}
+		// currently this hasn't been able to hit the endpoint replaceAppsV1NamespacedDeploymentStatus
+		_, err = dc.Resource(deploymentResource).Namespace(testNamespaceName).Update(context.TODO(), &testDeploymentUpdateUnstructured, metav1.UpdateOptions{}) //, "status")
```

Comment:
Why is `dc.Resource` being used instead of `f.ClientSet.AppsV1().Deployments`?

Reply:
I'm also curious why `Update` is being used here instead of the usual marshal-json-to-bytes-then-`Patch` approach I've seen in other tests from you?

Reply:
> Why is `dc.Resource` being used instead of `f.ClientSet.AppsV1().Deployments`?

This was meant to hit _replaceAppsV1NamespacedDeploymentStatus_, which to my research requires the DynamicClient - also why it's still remaining like this. Should I replace it with the default ClientSet, or should I leave it around for when the Endpoint could be added at a later date as a reminder?

Reply:
> I'm also curious why `Update` is being used here instead of the usual marshal-json-to-bytes-then-`Patch` approach I've seen in other tests from you?

Also because it's trying to hit the endpoint _replaceAppsV1NamespacedDeploymentStatus_.